### DISCUSSION TITLE
PDP Xbox One wired controller

### DISCUSSION THREAD

**Discussion by sabin1981** [cathery/sys-con#203] [Troubleshooting]
**Description of the controller**

It's the PDP Xbox One wired controller, this one is Verdant Green (and identifies as such in the PDP companion app on Xbox/Windows, so perhaps a unique VID/PID?) Wired connection via removable USB cable, no Bluetooth, no motion controls, no HD rumble just standard. Headphone port for audio out and microphone in, a "combo" key that - when held - adjusts volume control via dpad. No other special buttons or functions. Identifies in Windows as "Controller (Xbox One for Windows)" yet has additional information and functionality when opening the PDP Companion app.

When connected to the Switch a "USB Controller" shows up in the controller menu, but is non-functional. Nothing works, not sticks, triggers, nor buttons.

**Screenshot of the device**

![2020080311434700-35EB8ABB5248FD75298F28CD6232FBEA](https://user-images.githubusercontent.com/66366240/89175934-9759db00-d580-11ea-8d7b-a4cd663caf06.jpg)

**Additional context**

No other context but a preemptive thank you for the consideration. If I knew anything about modifying source to add the VID/PID and rebuilding, I'd happily do it myself but sadly I don't.

**Comment by cathery** [comment]
There's currently an unknown problem with latest xbox controller firmware causing issues. 
Try these things:
- Press the Xbox button so it starts blinking, then plug the controller into the switch
- Connect your controller through USB to your PC, then to your switch
- Use the Xbox Accessories app for Windows to reinstall controller firmware, then connect the controller to the switch

**Comment by sabin1981** [comment]
Unfortunately I can't press the Xbox button so it starts blinking, this is wired only.. no batteries, no power without USB, and connecting it to PC the little light above the Xbox guide button yet no light when connecting to Switch.

**Comment by cathery** [comment]
Yeah the first method requires power, it either requires batteries or having the USB cable just barely plugged so it sends power but not data.  This is for microsoft controllers anyway. Sadly I don't have the PDP controller to troubleshoot. If you could capture the packets sent and received between your PC and your controller it would be of help. You will need wireshark and usbpcap for this

**Comment by sabin1981** [comment]
> Yeah the first method requires power, it either requires batteries or having the USB cable just barely plugged so it sends power but not data. This is for microsoft controllers anyway. Sadly I don't have the PDP controller to troubleshoot. If you could capture the packets sent and received between your PC and your controller it would be of help. You will need wireshark and usbpcap for this

I've never used Wireshark before but I *think* I figured out the gist of it? I believe this is what you need? I ran a capture on the USBPCAP port that contained the Xbox One controller (I made sure to tick only that controller and nothing else) then pressed each button/trigger/stick/etc multiple times to ensure full capture. Is there anything I'm missing?

PDP Xbox One - Verdant Green.pcapng

[PDP Xbox One - Verdant Green.zip](https://github.com/cathery/sys-con/files/5015703/PDP.Xbox.One.-.Verdant.Green.zip)

**Comment by cathery** [comment]
In your packets i see the device "2.6.0" returning your controller's characteristics. The only thing I see it do is setting the configuration value to 1 (default is 0). I made a build that does the same for your controller. I deeply doubt it'll work, but it's worth a try.
[sys-con-pdp-test.zip](https://github.com/cathery/sys-con/files/5015938/sys-con-pdp-test.zip)

**Comment by sabin1981** [comment]
Sadly, you were right.. still no. I made a backup of the original sys-con, deleted it, installed the new build, rebooted ... still no go. The controller shows up in Control Setup as "USB Controller" but has no functionality, nor does the light shine above the XB logo like it does on stock XB controllers or when it is connected to PC,

Thank you, though!

**Comment by cathery** [comment]
Here's another build which doesn't send any packets to your controller whatsoever to closer mimic the packets on PC
[sys-con-pdp-test-2.zip](https://github.com/cathery/sys-con/files/5016024/sys-con-pdp-test-2.zip)

**Comment by sabin1981** [comment]
Once again, no dice. I'm guessing it has some kind of handshake with Windows/Xbox, the white light comes on when it is connected to PC or to an Xbox One, yet doesn't on Switch or my PS4.

**Comment by cathery** [comment]
Could be. Are you able to plug your controller to your switch in undocked mode? Microsoft pushed a controller firmware update to Xbox one controllers which somehow broke support with sys-con in docked, but somehow still working undocked.

**Comment by sabin1981** [comment]
No, sorry, I don't have a USB OTG adaptor nor do I have a USB-C to micro USB cable. I did try my USB-C to Micro-USB adaptor (that I use to charge my Galaxy S9 from standard Micro USB cables) but it wouldn't even work for my other controllers, which work via docked mode. 

Ah well, it's really no problem. I can always just get an official Xbox One controller :) Thank you for your help and input, it's greatly appreciated!

~edit~ 

Oh and for what it's worth; both a DualShock 4 (1st gen) and an official wireless Xbox One controller (my brother's) work flawlessly on the docked Switch via Sys-Con, when connected with a USB cable. It's just this PDP controller that's playing silly buggers.

**Comment by mbradshawdev** [comment]
Came here hoping for future PDP support and found this.
Tested PDP XBox One Wired Controller both docked and undocked, no luck.
Tested with two different OTG adapters both work with official Xbox360 wired, but not the PDP.
Also: The PDP does not work on Android 9 / Galaxy S8 while the official Xbox360 does.
Happy to provide any info if I am able

**Comment by Samekage09** [comment]
I was trying to connect the same controller and had the exact same issue
